src/flows/components/header/index.tsx

```diff
@@ -133,8 +108,25 @@ const FlowHeader: FC = () => {
       .catch(err => console.error('failed to delete share', err))
   }
 
-  const generateLink = () => {
+  const createAndGetReadOnlyToken = async (): Promise<Authorization> => {
+    const authName = `${flow.name} ${flow.createdAt}`
+
+    return await createReadOnlyAllAuthorization(orgID, authName)
+  }
+
+  const generateLink = async () => {
     setLinkLoading(RemoteDataState.Loading)
+    let token: Authorization
+
+    try {
+      token = await createAndGetReadOnlyToken()
```

Comment:
why do we need to store the token response?

Reply:
There's no particular reason. I will remove it.

Reply:
K, we also never set the linkLoading to done. Do we want to do that?